### PR TITLE
support using any value from owned resource in path with {{.owned.xxx…

### PR DESCRIPTION
**Pull request by kuanf**
….xxx.xx}}

#38 

There is a problem with escaping dot (.) in jsonpath for ORM in general. It is addressed in separate issue #40

### CODE REVIEW

**Comment by ading1977** [comment]
Can you provide some sample CR yamls reflecting this changes? Should it be in https://github.com/turbonomic/orm/blob/master/config/samples?

**Comment by kuanf** [comment]
> Can you provide some sample CR yamls reflecting this changes? Should it be in https://github.com/turbonomic/orm/blob/master/config/samples?

updated the orm sample to show the syntax. particularly for this feature, the {{.owned.labels.appname}} gets the value of a label in deployment (e.g. "mydb" in the example below and use that in owner path)
 
```yaml
metadata:
  labels:
    appname: mydb
```

File: registry/orm.go [hunk 1]
```diff
@@ -439,7 +482,80 @@ func populatePatterns(parameters map[string][]string, pattern devopsv1alpha1.Pat
...
+	if start == -1 {
+		return path, false, nil
+	}
+
+	end := strings.Index(path, "}}")
+	if end == -1 {
+		return path, false, errors.New(errorMessageSyntaxError + path)
+	}
+
+	full := path[start : end+len(predefinedquoteend)]
```
Comment (tian-ma) [inline registry/orm.go:532]: "full"?

File: registry/orm.go [hunk 2]
```diff
@@ -413,11 +443,24 @@ func (or *ResourceMappingRegistry) retrieveObjectEntryForOwnerAndORM(owner corev
 	return retrieveObjectEntryForObjectAndORMFromRegistry(or.ownerRegistry, owner, orm)
 }
 
-func populatePatterns(parameters map[string][]string, pattern devopsv1alpha1.Pattern) []devopsv1alpha1.Pattern {
+func processshorthandvariables(str string) string {
```
Comment (tian-ma) [inline registry/orm.go:446]: camel case?

**Review by tian-ma — CHANGES_REQUESTED** [review]

File: registry/orm.go (hunk 1 above)
Comment (tian-ma) [reply registry/orm.go:532]: I find all the variable names problematic in terms of readability. 
They resemble functions and keywords from various programming languages and not descriptive (self documenting code)
```
path =>              rawObjectPath

content =>           objectPath 
```

```
if -1==strings.Index(objPathContentKey) {
   return ....
}
objectPath = strings.NewReplacer("{","","}","").Replace(rawObjectath)
```

**Review by tian-ma — CHANGES_REQUESTED** [review]

File: registry/orm.go (hunk 1 above)
Comment (kuanf) [reply registry/orm.go:545]: I added comment to explain the function and the variables defined there. Let me know if you need more clarification.

**Review by tian-ma — APPROVED** [review]

**Pull request merged**